### PR TITLE
Adding unittests for logger/logger.go.

### PR DESCRIPTION
**Pull request by omaciel**
# Description

Adding unittests to `logger/logger.go`.  Changed `logger.InitLogger` function to accept an `io.Writer` to make it easy to test it and updated all occurrences to pass `os.Stdout` as the default.

Fixes # (issue)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Comment by omaciel** [comment]
@ldjebran what do you think?

**Comment by mgold1234** [comment]
fail in cmd/db/updDb/set_account_on_device.go:5: File is not `goimports`-ed (goimports)
  import (
  logger/logger_test.go:8: File is not `goimports`-ed (goimports)
  	"github.com/redhatinsights/edge-api/config"

File: logger/logger_test.go [hunk 1]
```diff
@@ -3,19 +3,89 @@
...
+	myLog := log.WithFields(log.Fields{"app": "edge", "service": "images"})
+	myLog.Info(want)
+
+	logger.FlushLogger()
+	got := buffer.String()
+	assert.Contains(t, got, want)
+}
+
+func TestInitLogger(t *testing.T) {
+	tests := []struct {
```
Comment (ldjebran) [inline logger/logger_test.go:53]:
```go
originalLogLevel := config.Get().LogLevel
// defre restore log level
defer func (level) {
     config.Get().LogLevel = originalLogLevel
}(originalLogLevel)

```

File: logger/logger_test.go [hunk 2]
```diff
@@ -3,19 +3,89 @@
...
+	want := "Test flushing log messages"
+	myLog := log.WithFields(log.Fields{"app": "edge", "service": "images"})
+	myLog.Info(want)
+
+	logger.FlushLogger()
+	got := buffer.String()
+	assert.Contains(t, got, want)
+}
+
+func TestInitLogger(t *testing.T) {
```
Comment (ldjebran) [inline logger/logger_test.go:52]:
```go
originalLogLevel := config.Get().LogLevel
// defre restore log level
defer func (level) {
     config.Get().LogLevel = originalLogLevel
}(originalLogLevel)
```

File: logger/logger_test.go [hunk 3]
```diff
@@ -3,19 +3,89 @@
...
+		},
+		{
+			name:     "Use Info log level",
+			logLevel: "DEBUG",
+			level:    log.InfoLevel,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			originalLogLevel := config.Get().LogLevel
```
Comment (ldjebran) [inline logger/logger_test.go:76]: we remove this , if making recommended changes above

File: logger/logger_test.go [hunk 4]
```diff
@@ -3,19 +3,89 @@
...
+			writer := &bytes.Buffer{}
+
+			logger.InitLogger(writer)
+
+			log.SetLevel(tt.level)
 
 			logger.FlushLogger()
+			assert.Equal(t, tt.level, log.GetLevel())
+
+			config.Get().LogLevel = originalLogLevel
```
Comment (ldjebran) [inline logger/logger_test.go:88]: we remove this , if making recommended changes above

File: logger/logger_test.go [hunk 5]
```diff
@@ -3,19 +3,89 @@
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			originalLogLevel := config.Get().LogLevel
+			config.Get().LogLevel = tt.logLevel
+
+			writer := &bytes.Buffer{}
+
+			logger.InitLogger(writer)
+
+			log.SetLevel(tt.level)
```
Comment (ldjebran) [inline logger/logger_test.go:83]: do we need this line ?
 may be it should be removed , as it should be set by InitLogger

**Review by ldjebran — COMMENTED** [review]
@omaciel  I like the idea :) , left some comments

File: logger/logger_test.go (hunk 2 above)
Comment (omaciel) [reply logger/logger_test.go:52]: Done

File: logger/logger_test.go [hunk 6]
```diff
@@ -3,19 +3,87 @@
...
+			defer func(level string) {
+				config.Get().LogLevel = level
+			}(originalLogLevel)
+
+			config.Get().LogLevel = tt.logLevel
+
+			writer := &bytes.Buffer{}
 
+			logger.InitLogger(writer)
 			logger.FlushLogger()
```
Comment (ldjebran) [inline logger/logger_test.go:86]: @omaciel we lost the assert :) 
```go
assert.Equal(t, tt.level, log.GetLevel())
```

Comment (omaciel) [reply logger/logger_test.go:87]: hahaha added it back. Sorry.

**Review by ldjebran — APPROVED** [review]
/lgtm

**Comment by ldjebran** [comment]
remove do not merge as automation is ok now

File: cmd/db/updDb/set_account_on_device.go [hunk 7]
```diff
@@ -3,16 +3,19 @@
...
 	log "github.com/sirupsen/logrus"
 	"gorm.io/gorm"
+
+	"github.com/redhatinsights/edge-api/config"
 )
 
 func main() {
 	config.Init()
-	l.InitLogger()
+	l.InitLogger(os.Stdout)
```
Comment (de1987) [inline cmd/db/updDb/set_account_on_device.go:18]: I'm concerned that it might cause excessive number of logs.
@omaciel could it happen?

Comment (omaciel) [reply cmd/db/updDb/set_account_on_device.go:18]: The default was logging to stdout. I merely moved it from the module and turned it into a parameter

**Review by de1987 — APPROVED** [review]

**Comment by omaciel** [comment]
@Mergifyio rebase

**Pull request merged**